mlir/include/Quantum/IR/QuantumOps.td

```diff
@@ -203,7 +204,8 @@ def CustomOp : Gate_Op<"custom", [NoMemoryEffect, AttrSizedOperandSegments, Diff
     let arguments = (ins
         Variadic<F64>:$params,
         Variadic<QubitType>:$in_qubits,
-        StrAttr:$gate_name
+        StrAttr:$gate_name,
+        OptionalAttr<BoolAttr>:$adjoint
```

Comment:
Might also be simpler to have an optional unit attr, since the presence of the attribute is sufficient to distinguish between two states.

Reply:
Makes sense, will do.